src/Time/StepChoosers/StepChooser.hpp

```diff
@@ -28,6 +32,42 @@ namespace StepChoosers {
 /// These can be passed in a list to the template argument of
 /// StepChooser to choose which StepChoosers can be constructed.
 namespace Registrars {}
+
+CREATE_HAS_TYPE_ALIAS(slab_choosers)
+CREATE_HAS_TYPE_ALIAS_V(slab_choosers)
+CREATE_GET_TYPE_ALIAS_OR_DEFAULT(slab_choosers)
+CREATE_HAS_TYPE_ALIAS(step_choosers)
+CREATE_HAS_TYPE_ALIAS_V(step_choosers)
+CREATE_GET_TYPE_ALIAS_OR_DEFAULT(step_choosers)
+CREATE_HAS_TYPE_ALIAS(compute_tags)
+CREATE_HAS_TYPE_ALIAS_V(compute_tags)
+CREATE_GET_TYPE_ALIAS_OR_DEFAULT(compute_tags)
+CREATE_HAS_TYPE_ALIAS(simple_tags)
+CREATE_HAS_TYPE_ALIAS_V(simple_tags)
+CREATE_GET_TYPE_ALIAS_OR_DEFAULT(simple_tags)
+
+/// Designation for the context in which a step chooser may be used
+enum class UsableFor { OnlyLtsStepChoice, OnlySlabChoice, AnyStepChoice };
+
+template <typename Metavariables>
+using step_chooser_compute_tags =
+    tmpl::remove_duplicates<tmpl::flatten<tmpl::transform<
+        typename StepChooser<tmpl::remove_duplicates<tmpl::append<
+            get_step_choosers_or_default_t<Metavariables, tmpl::list<>>,
+            get_slab_choosers_or_default_t<Metavariables, tmpl::list<>>>>>::
+            creatable_classes,
+        tmpl::bind<get_compute_tags_or_default_t, tmpl::_1,
```

Comment:
This didn't end up working. I think it is because the template specialization in the implementation of the `get_..._or_default` involves a `bool` parameter for deciding between specializations, so doesn't fit the brigand form for lambdas being template template paramters with only type packs.

Reply:
We should fix that, but not right now.